python/ray/dag/compiled_dag_node.py

```diff
@@ -384,19 +394,47 @@ def _get_or_compile(
             # Create an output buffer on the actor.
             assert task.output_channel is None
             if isinstance(task.dag_node, ClassMethodNode):
+                readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
+                assert len(readers) == 1
+                if isinstance(readers[0].dag_node, MultiOutputNode):
+                    readers = [None]
```

Comment:
This line doesn't seem necessary.

Can you also add a comment to this if-block explaining the case?

Reply:
Done.